ipa-core/src/cli/crypto.rs

```diff
@@ -181,60 +180,88 @@ pub async fn decrypt_and_reconstruct(args: DecryptArgs) -> Result<(), BoxError>
             .create_new(true)
             .open(args.output_file)?,
     );
-
-    for (dec_report1, (dec_report2, dec_report3)) in
-        decrypted_reports1.zip(decrypted_reports2.zip(decrypted_reports3))
+    let mut first_error = Ok(());
+    for (idx, (dec_report1, (dec_report2, dec_report3))) in decrypted_reports1
+        .zip(decrypted_reports2.zip(decrypted_reports3))
+        .enumerate()
     {
-        let timestamp = [
-            dec_report1.timestamp,
-            dec_report2.timestamp,
-            dec_report3.timestamp,
-        ]
-        .reconstruct()
-        .as_u128();
-
-        let match_key = [
-            dec_report1.match_key,
-            dec_report2.match_key,
-            dec_report3.match_key,
-        ]
-        .reconstruct()
-        .as_u128();
-
-        // these aren't reconstucted, so we explictly make sure
-        // they are consistent across all three files, then set
-        // it to the first one (without loss of generality)
-        assert_eq!(dec_report1.event_type, dec_report2.event_type);
-        assert_eq!(dec_report2.event_type, dec_report3.event_type);
-        let is_trigger_report = dec_report1.event_type == EventType::Trigger;
-
-        let breakdown_key = [
-            dec_report1.breakdown_key,
-            dec_report2.breakdown_key,
-            dec_report3.breakdown_key,
-        ]
-        .reconstruct()
-        .as_u128();
-
-        let trigger_value = [
-            dec_report1.trigger_value,
-            dec_report2.trigger_value,
-            dec_report3.trigger_value,
-        ]
-        .reconstruct()
-        .as_u128();
-
-        writeln!(
-            writer,
-            "{},{},{},{},{}",
-            timestamp,
-            match_key,
-            u8::from(is_trigger_report),
-            breakdown_key,
-            trigger_value,
-        )?;
+        match (dec_report1, dec_report2, dec_report3) {
```

Comment:
it may make sense to check each for errors explicitly up top, so that we make sure we log all 3. we'll want to make sure that all 3 are not decryptable. the suggestion is just a sketch, don't merge it directly.

```suggestion
        if dec_report1.is_err() { # log here }
        if dec_report2.is_err() { # log here }
        if dec_report3.is_err() { # log here }
        match (dec_report1, dec_report2, dec_report3) {
            (Ok(dec_report1), Ok(dec_report2), Ok(dec_report3)) => { ... }
            _ => { ... }
        }
```